### PR TITLE
Annotate metadata calls with 'nogil'

### PR DESCRIPTION
**Pull request by ihnorton**

### CODE REVIEW

**Review by nguyenv — APPROVED** [review]

File: tiledb/libmetadata.pyx [hunk 1]
```diff
@@ -328,16 +342,19 @@ cdef class Metadata:
...
-        cdef int32_t rc = tiledb_array_has_metadata_key(
-            ctx_ptr,
-            array_ptr,
-            <const char*>key_utf8,
-            &value_type,
-            &has_key,
-        )
+        cdef int32_t rc = TILEDB_OK
+        with nogil:
+            tiledb_array_has_metadata_key(
```
Comment (gsakkis) [inline tiledb/libmetadata.pyx:351]: Shouldn't this assign `rc`?

File: tiledb/libmetadata.pyx [hunk 2]
```diff
@@ -411,7 +431,9 @@ cdef class Metadata:
             tiledb_array_t* array_ptr = (<Array>self.array).ptr
             uint64_t num
 
-        cdef int32_t rc = tiledb_array_get_metadata_num(ctx_ptr, array_ptr, &num)
+        cdef int32_t rc = TILEDB_OK
+        with nogil:
+            tiledb_array_get_metadata_num(ctx_ptr, array_ptr, &num)
```
Comment (gsakkis) [inline tiledb/libmetadata.pyx:436]: Shouldn't this assign rc?

File: tiledb/libmetadata.pyx (hunk 1 above)
Comment (ihnorton) [reply tiledb/libmetadata.pyx:351]: 👍

File: tiledb/libmetadata.pyx (hunk 2 above)
Comment (ihnorton) [reply tiledb/libmetadata.pyx:436]: Thanks

**Pull request merged**